_posts/2021-11-01-Fortran-Newsletter-November-2021.md

```diff
@@ -0,0 +1,246 @@
+    * Move the parsers, printers and builders from the TableGen file to the .cpp file
+    * Update fir.alloca op - Add pinned attributes and specific builders
+    * Add ops: fir.char_convert and fir.array_modify
+    * Add passes: external name interop, affine promotion, affine demotion, character conversion, abstract result conversion, cfg conversion
+    * Add fir.convert canonicalization patterns
+    * Add the DoLoopHelper
+    * Add IfBuilder and utility functions
+    * Add FIRBuilder utility functions
+    * Add character utility functions in FIRBuilder
+    * Add Character helper
+    * Add utility function to FIRBuilder and MutableBox
+    * Add substring to fir.slice operation
+    * Avoid slice with substr in fir.array_load, fir.array_coor and fir.array_merge_store
+* Driver
+    * Error if uuidgen is not installed
+    * Fix erroneous `&`
+    * Add actions that execute despite semantic errors
+* flang-omp-report 
+    * replace std::vector's with llvm::SmallVector
+    * Switch from std::string to StringRef (where possible)
+    * replace std::map with llvm::DenseMap
+* Make builtin types more easily accessible; use them
+* Fix test regression from SQRT folding
+* Fold FINDLOC, MAXLOC, MINLOC, LGE/LGT/LLE/LLT, BTEST intrinsic functions
+* Take into account SubprogramDetails in GetInterfaceSymbol
+* Add debug dump method to evaluate::Expr and semantics::Symbol
+* Add a wrapper for Fortran main program
+* Improve runtime interface with C99 complex
+* Better error recovery for missing THEN in ELSE IF
+* Define IEEE_SCALB, IEEE_NEXT_AFTER, IEEE_NEXT_DOWN, IEEE_NEXT_UP
+* Catch mismatched parentheses in prescanner
+* Error checking for IBCLR/IBSET and ISHFT/SHIFT[ALR]
+* Document behavior for nonspecified/ambiguous cases
+* Add two negative tests for needExternalNameMangling
+* Expunge bogus semantic check for ELEMENTAL without dummies
+* Admit NULL() in generic procedure resolution cases
+* Fix bogus folding error for ISHFT(x, negative)
+* Emit unformatted headers & footers even with RECL=
+* Enforce rest of semantic constraint C919
+* Extension to distinguish specific procedures
+* Support NAMELIST input of short arrays
+* Fix generic resolution case
+* Speed common runtime cases of DOT_PRODUCT & MATMUL
+* Fix crash on empty formatted external READs
+* Extension: allow tabs in output format strings
+* Fix DOT_PRODUCT for logical
+* Fix NAMELIST input bug with multiple subscript triplets
+* Support legacy usage of 'A' edit descriptors for integer & real
+
+Call notes are recorded and available upon request [here](https://docs.google.com/document/d/10T-S2J3GrahpG4Ooif93NSTz2zBW0MQc_RlwHi0-afY). Please contact Alexis Perry-Holby at aperry@lanl.gov for document access.
+
+### LFortran
+
+
+We are looking for new contributors. Please do not hesitate to contact us if
+you are interested. We will help you get up to speed.
+
+
+## Events
+
+* We had our 17th Fortran Monthly call on September 19.
```

Comment:
```suggestion
* We had our 19th Fortran Monthly call on October 19.
```

(I think 19th call is correct because we have 18 videos in the YT channel and the first meeting was not recorded)